File: backend/mr-db-sql/src/main/java/io/featurehub/db/listener/FeatureUpdateBySDKApi.java
package io.featurehub.db.listener;

import io.featurehub.db.api.RolloutStrategyValidator;
import io.featurehub.mr.model.FeatureValue;
import io.featurehub.mr.model.FeatureValueType;

import java.util.function.Function;

public interface FeatureUpdateBySDKApi {
  void updateFeature(String sdkUrl, String envId, String featureKey, boolean updatingValue, Function<FeatureValueType, FeatureValue> buildFeatureValue)
    throws RolloutStrategyValidator.InvalidStrategyCombination;
}

File: backend/mr-db-services/src/main/kotlin/io/featurehub/db/listener/FeatureUpdateBySDKApi.kt
package io.featurehub.db.listener

import io.featurehub.db.api.RolloutStrategyValidator
import io.featurehub.events.CloudEventReceiverRegistry
import io.featurehub.mr.model.FeatureValue
import io.featurehub.mr.model.FeatureValueType
import jakarta.inject.Inject
import java.util.*
import java.util.function.Function

interface FeatureUpdateBySDKApi {
    @Throws(RolloutStrategyValidator.InvalidStrategyCombination::class)
    fun updateFeatureFromTestSdk(
        sdkUrl: String,
        envId: UUID,
        featureKey: String,
        updatingValue: Boolean,
        updatingLock: Boolean,
        buildFeatureValue: Function<FeatureValueType, FeatureValue>
    )
}

